class.jetpack.php

```diff
@@ -4622,6 +4622,12 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 		// Get affiliate code and add it to the URL
 		$url = Jetpack_Affiliate::init()->add_code_as_query_arg( $url );
 
+		// Ensure that class to get the partner subsidiary id is loaded.
+		if ( ! class_exists( 'Jetpack_Partner_Subsidiary' ) ) {
```

Comment:
Instead of doing this. Have we though about implement it as a filter instead?

Reply:
It's implemented as a filter (and an option) here:

https://github.com/Automattic/jetpack/blob/63d5b6f979e6699c757581149bc3dd7305244c08/class.jetpack-partner-subsidiary.php#L66-L74

Reply:
What I was trying to get to is instead of doing 
```
I think instead of doing 

$url = Jetpack_Affiliate::init()->add_code_as_query_arg( $url );
$url = Partner::init()->add_subsidiary_id_as_query_arg( $url );
return $raw ? esc_url_raw( $url ) : esc_url( $url );
```

I think we should do something like the following. 
```
$url = $raw ? esc_url_raw( $url ) : esc_url( $url )
return apply_filters( 'jetpack_build_connection_url', $url, $raw );
```
and then 
on add the filters after all the plugin have loaded. 
```
add_filter( 'jetpack_build_connection_url', array( 'Jetpack_Affiliate', 'add_code_as_query_arg' );
add_filter( 'jetpack_build_connection_url', array( 'Partner', 'add_subsidiary_id_as_query_arg' );
```